watertap/unit_models/electrodialysis_1D.py

```diff
@@ -778,6 +1070,497 @@ def eq_recovery_mass_H2O(self, t):
+            if (
+                self.config.limiting_current_density_method
+                == LimitingCurrentDensityMethod.InitialValue
+            ):
+                return self.current_dens_lim_x[t, x] == (
+                    self.config.limiting_current_density_data
+                    * pyunits.amp
+                    * pyunits.meter**-2
+                    / sum(
+                        self.diluate.properties[t, 0].conc_mol_phase_comp["Liq", j]
+                        for j in self.cation_set
+                    )
+                    * sum(
+                        self.diluate.properties[t, x].conc_mol_phase_comp["Liq", j]
+                        for j in self.cation_set
+                    )
+                )
+
+        @self.Constraint(
+            self.membrane_set,
+            self.electrode_side_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            self.config.property_package.ion_set,
+            doc="Establish relationship between interfacial concentration polarization ratio and current density",
+        )
+        def eq_conc_polarization_ratio(self, mem, side, t, x, j):
+            if (mem == "cem" and side == "cathode_left") or (
+                mem == "aem" and side == "anode_right"
+            ):
+                return self.conc_mem_surf_mol_x[
+                    mem, side, t, x, j
+                ] / self.concentrate.properties[t, x].conc_mol_phase_comp["Liq", j] == (
+                    1 + self.current_density_x[t, x] / self.current_dens_lim_x[t, x]
+                )
+            else:
+                return self.conc_mem_surf_mol_x[
+                    mem, side, t, x, j
+                ] / self.diluate.properties[t, x].conc_mol_phase_comp["Liq", j] == (
+                    1 - self.current_density_x[t, x] / self.current_dens_lim_x[t, x]
+                )
+
+        @self.Constraint(
+            self.membrane_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            doc="Calculate the total non-ohmic potential across the two diffusion layers of an iem.",
+        )
+        def eq_potential_nonohm_dl(self, mem, t, x):
+            if mem == "cem":
+                return self.potential_nonohm_dl_x[mem, t, x] == (
+                    Constants.gas_constant
+                    * self.diluate.properties[t, x].temperature
+                    / Constants.faraday_constant
+                    * (
+                        sum(
+                            self.diluate.properties[t, x].trans_num_phase_comp["Liq", j]
+                            / self.config.property_package.charge_comp[j]
+                            for j in self.cation_set
+                        )
+                        + sum(
+                            self.diluate.properties[t, x].trans_num_phase_comp["Liq", j]
+                            / self.config.property_package.charge_comp[j]
+                            for j in self.anion_set
```

Comment:
out of curiosity, could this be reduced to a single summation across `self.ion_set`?